src/app/frontend/replicasetlist/replicasetlist_controller.js

```diff
@@ -28,10 +29,24 @@ export default class ReplicaSetListController {
    */
   constructor($state, replicaSets) {
     /** @export {!Array<backendApi.ReplicaSet>} */
-    this.replicaSets = replicaSets.replicaSets;
+    this.replicaSets = replicaSets.replicaSets || [];
 
     /** @private {!ui.router.$state} */
     this.state_ = $state;
+
+    this.initState_();
```

Comment:
Hmm... I'm not sure whether this functionality belongs to the controller of replica set list. What do you think on checking this condition in onEnter function in the replicasetlist_state and transition to zerostate there?